### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by bmildren**
##### SUMMARY
Migrated the existing proxysql modules into the collection, and added a basic role.  Added initial config for the collection, including some initial integration tests, and github actions workflows.

### CODE REVIEW

File: .github/workflows/ansible-test-plugins.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      - name: Set up Python ${{ matrix.python }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Install ansible-base (${{ matrix.ansible }})
+        run: pip install https://github.com/ansible/ansible/archive/${{ matrix.ansible }}.tar.gz --disable-pip-version-check
+
+      - name: Run sanity tests
+        run: ansible-test sanity --docker -v --color --python ${{ matrix.python }}
```
Comment (felixfontein) [inline .github/workflows/ansible-test-plugins.yml:48]:
```suggestion
        run: ansible-test sanity --docker -v --color
```
This way, all Python versions known to that ansible-test version will be used. (Remove `python` from the text matrix.)

File: .github/workflows/ansible-test-plugins.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  proxysql_version_file: "./ansible_collections/community/proxysql/tests/integration/targets/setup_proxysql/defaults/main.yml"
+
+jobs:
+  sanity:
+    name: "Sanity (Python: ${{ matrix.python }}, Ansible: ${{ matrix.ansible }})"
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ansible:
+          - stable-2.10
```
Comment (felixfontein) [inline .github/workflows/ansible-test-plugins.yml:27]: Why not also use stable-2.9 here?

File: README.md [hunk 3]
```diff
@@ -1,58 +1,44 @@
...
-<!-- Add CI and code coverage badges here. Samples included below. -->
-[![CI](https://github.com/ansible-collections/REPONAMEHERE/workflows/CI/badge.svg?event=push)](https://github.com/ansible-collections/REPONAMEHERE/actions) [![Codecov](https://img.shields.io/codecov/c/github/ansible-collections/REPONAMEHERE)](https://codecov.io/gh/ansible-collections/REPONAMEHERE)
+# ProxySQL collection for Ansible
+[![CI](https://github.com/ansible-collections/community.proxysql/workflows/CI/badge.svg?event=push)](https://github.com/ansible-collections/community.proxysql/actions) [![Codecov](https://img.shields.io/codecov/c/github/ansible-collections/community.proxysql)](https://codecov.io/gh/ansible-collections/community.proxysql)
 
 <!-- Describe the collection and why a user would want to use it. What does the collection do? -->
 
-## Tested with Ansible
-
-<!-- List the versions of Ansible the collection has been tested with. Must match what is in galaxy.yml. -->
```
Comment (felixfontein) [inline README.md:12]: I'd keep this section.

File: README.md [hunk 4]
```diff
@@ -1,58 +1,44 @@
...
-
-## External requirements
-
-<!-- List any external resources the collection depends on, for example minimum versions of an OS, libraries, or utilities. Do not list other Ansible collections here. -->
```
Comment (felixfontein) [inline README.md:16]: In case you need any SDK or other Python requirement, you should mention it here.

File: galaxy.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+name: proxysql
+version: 1.0.0
+readme: README.md
+authors:
+  - Ben Mildren (@bmildren)
+description: ProxySQL collection for Ansible
+license_file: LICENSE
+tags:
+  - database
+  - mysql
```
Comment (felixfontein) [inline galaxy.yml:11]:
```suggestion
  - mysql
  - proxysql
```

File: .github/workflows/ansible-test-plugins.yml (hunk 2 above)
Comment (bmildren) [reply .github/workflows/ansible-test-plugins.yml:27]: cool ok I'll add that in

File: README.md (hunk 3 above)
Comment (bmildren) [reply README.md:12]: cool ok

File: README.md (hunk 4 above)
Comment (bmildren) [reply README.md:16]: ack

File: galaxy.yml (hunk 5 above)
Comment (bmildren) [reply galaxy.yml:11]: 👍

File: .github/workflows/ansible-test-plugins.yml (hunk 1 above)
Comment (bmildren) [reply .github/workflows/ansible-test-plugins.yml]: I wonder if it's best to be explicit here? (to ensure we're testing the exact versions we expect)

**Pull request merged**